### DISCUSSION TITLE
Modularisation - redundant dependencies

### DISCUSSION THREAD

**Discussion by Przeszaf** [pointfreeco/isowords#148] [Q&A] [answered]
Hey, I've been working on modularising my app and found out that, if a child dependency imports something, then parent doesn't need to.

I noticed that there are some redundant imports in isowords and was wondering if I should track and remove them in my app or are they fine to stay there?

One example:
`DatabaseClient` has a dependency on `SharedModels`.
`DemoMiddleware` has a dependency on both `DatabaseClient` and `SharedModels` therefore `SharedModels` dependency could be removed from `DemoMiddleware`.

Would it make any difference if it was removed in terms of compile time or anything at all?

**Comment by stephencelis** [comment, marked as answer]
@Przeszaf We've preferred making even transitive dependencies explicit, but it's up to you! One thing about making them explicit is if you remove a dependency from `ModuleA` that is depended on by `ModuleB`, but `ModuleB` depended on that dependency transitively, you must also update `ModuleB`'s dependencies.
